src/components/Ads/index.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from 'react'
+import { View, Text, StyleSheet, Platform } from 'react-native'
+import { AdMobBanner } from 'react-native-admob'
+
+const Ads = props => {
+  let {
+    iosAdID,
+    andAdID,
+    size,
+    iosAppIDGlobal,
+    iosAppID,
+    andAppIDGlobal,
+    andAppID,
+  } = props
+
+  if (
+    (Platform.OS === 'ios' && (!iosAppIDGlobal || !iosAppID)) ||
+    (Platform.OS === 'android' && (!andAppIDGlobal || !andAppID))
+  ) {
+    const platformPretty = Platform.OS === 'ios' ? 'iOS' : 'Android'
+    return (
+      <View style={styles.errorView}>
+        <Text style={styles.errorText}>
+          Component Setup Incomplete. {platformPretty} App ID was left blank.
+        </Text>
+      </View>
+    )
+  }
+
+  const [adID, setAdId] = useState('')
+
+  useEffect(() => {
+    if (Platform.OS === 'ios') {
+      if (iosAdID) setAdId(iosAdID.replace(/\s/g, ''))
+    } else if (Platform.OS === 'android') {
+      if (andAdID) setAdId(andAdID.replace(/\s/g, ''))
+    }
+  }, [iosAdID, andAdID])
```

Comment:
Need to update the `set` tuple name to match the call. I think `Platform.OS` is required in the useEffect dependencies as well.

```suggestion
  const [adID, setAdID] = useState('')

  useEffect(() => {
    if (Platform.OS === 'ios') {
      if (iosAdID) setAdId(iosAdID.replace(/\s/g, ''))
    } else if (Platform.OS === 'android') {
      if (andAdID) setAdId(andAdID.replace(/\s/g, ''))
    }
  }, [Platform.OS, iosAdID, andAdID])
```

Reply:
Why add `Platform.OS`? There is no situation in which that will change

Reply:
The `useEffect` dependency array will only fire if one of the included dependencies changes. While unlikely, we do have conditional statements that rely on `Platform.OS`, and if that changes we'll want to update our `AdID`.